### ISSUE TITLE
not working in plain react with vite

### ISSUE THREAD

**Issue by ahmadghoniem** [Nutlope/nsfw-filter#1]
![image](https://github.com/Nutlope/nsfw-filter/assets/60299751/b5550bda-2940-4454-9dc7-d5edb821e410)

Hello hassan, really love your work and wanted to give it a try however it doesn't work with plain react
is that example that you posted was for a server component with nextjs? maybe that's why it's not working on my end?

here's the error from the console too

![image](https://github.com/Nutlope/nsfw-filter/assets/60299751/8c0eb5bf-971f-44f9-ab4d-7fcf31dcacf0)

**Comment by ahmadghoniem** [comment]
got it fixed!
will comment with the solution to help users who will run into the same issue in the future

**Comment by Nutlope** [comment]
Thank you, please do!

**Comment by ahmadghoniem** [comment]
The problem was that nsfw-filter was using multiple core node modules that are not supported by the browser
which will result in an error for example using fs node module
`Module "fs" has been externalized for browser compatibility. Cannot access "fs.readFile" in client code.`
[more on this]( https://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility)

in [restorePhotos](https://github.com/Nutlope/restorePhotos/blob/main/pages/restore.tsx#L50) nsfw-filter was used in a server component which won't result in any type of errors as the core node modules are available because the react component is running on a nodejs server

however when you try to use nsfw-filter in a browser you will need to polyfill those node modules for browser support the ones that used by nsfw-filter are
`["stream", "assert", "events", "zlib", "util", "buffer"]`
you can do that one by one which is okay for example downloading[ stream module](https://www.npmjs.com/package/stream) through npm  
and the same goes for the others
however you might run into a problem that a few of those npm packages don't support the browser and are ported out of node and put into an npm package
like [zlib](https://www.npmjs.com/package/zlib) and you will need to find a zlib module that supports the browser
like [browserify-zlib](https://www.npmjs.com/package/browserify-zlib)
if you want to save your self the hustle you can use 
[vite-plugin-node-polyfills](https://www.npmjs.com/package/vite-plugin-node-polyfills
)
and use it like this  in your vite.config.json   
```
nodePolyfills({
      // To add only specific polyfills, add them here. If no option is passed, adds all polyfills
      include: ["path", "stream", "assert", "events", "zlib", "util", "buffer"],
    }),
```

bonus: i've found out that you can polyfill path to support path aliases for vite when you deploy your could on the cloud for example vercel

so you can use `@ ` 
`import { cn } from "@/lib/utils";`

**Comment by Nutlope** [comment]
I appreciate the context, just merged #3!
